src/patterns/pages/workspace/data/workspace.service.ts

```diff
@@ -0,0 +1,214 @@
+import { Injectable } from '@angular/core';
+import { SampleOppData } from './datasource';
+import { Observable } from 'rxjs';
+
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WorkspaceService {
+
+  constructor() { }
+
+  //Will pass in 
+  getData(filter: filter) {
+
+    let data = <Opportunity[]>SampleOppData;
+    let ob = Observable.of(data);
+    ob = this.filter(ob, filter);
+    let pageSize = 10;
+
+    return ob;
+  }
+
+
+  private filter(data: Observable<Opportunity[]>, filter: filter) {
+
+    if (filter.type) {
+      data = data.map(projects => projects.filter(proj => proj.data.type === filter.type));
+    }
+
+    if (filter.status) {
+      data = data.map(projects => projects.filter(proj => proj.status === filter.status));
+    }
+
+    if (filter.title) {
+      data = data.map(projects => projects.filter(proj => proj.data.title.indexOf(filter.title) !== -1));
+    }
+
+    return data;
+  }
+
+  sort(data: Opportunity[], filter: filter) {
```

Comment:
@thomastighe Is this empty because its WIP? 

If the implementor needs to write the method, we should probably look at making this `abstract` or implement an interface.

Reply:
Yet again this is only the sample service for the pattern to demo our page.